src/commands/init/generateTemplate/collect.js

```diff
@@ -0,0 +1,23 @@
+export default async function(data, metadata, done) {
+    const keys = Object.getOwnPropertyNames(data);
+
+    for (const key of keys) {
+        await unwrap(metadata, key, data[key]);
+    }
+
+    done();
+}
+
+async function unwrap(metadata, key, dataProvider) {
+    /* eslint-disable no-param-reassign */
+    if (typeof dataProvider === 'function') {
+        try {
+            metadata[key] = await dataProvider();
```

Comment:
What do you think about passing in the `metadata` here, into `dataProvider()`?

Then we could get access to the answers that the user has answered.

```js
// Provide metadata
await dataProvider(metadata);

// Later we can use it
"recentChanges":  (answers) => fetch("https://some-changelog.org/recent")
```

If we do this change we should also add a comment about it in the documentation.